servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java

```diff
@@ -492,12 +485,15 @@ private ImmutableLogEntry commitToLogEntry(boolean fetchAll, Commit commit) {
   private static CommitMeta enhanceCommitMeta(
       Hash hash, CommitMeta commitMeta, List<Hash> additionalParents) {
     ImmutableCommitMeta.Builder updatedCommitMeta = commitMeta.toBuilder().hash(hash.asString());
-    if (additionalParents != null && additionalParents.size() == 1) {
-      // Only add the 1st commit ID. The MERGE_PARENT_PROPERTY was introduced for compatibility
-      // with older clients. There is currently only one use case for the property: exposing the
-      // commit ID of the merged commit.
-      updatedCommitMeta.putProperties(
-          CommitMeta.MERGE_PARENT_PROPERTY, additionalParents.get(0).asString());
+    if (additionalParents != null && !additionalParents.isEmpty()) {
+      additionalParents.forEach(p -> updatedCommitMeta.addParentCommitHashes(p.asString()));
+      if (additionalParents.size() == 1) {
+        // Only add the 1st commit ID. The MERGE_PARENT_PROPERTY was introduced for compatibility
```

Comment:
nit: this comment looks a bit fuzzy... we have only one element in this case to begin with :thinking: